services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -506,3 +511,39 @@ async def try_to_remove_network(network_name: str) -> None:
             return await network.delete()
         except aiodocker.exceptions.DockerError:
             log.warning("Could not remove network %s", network_name)
+
+
+async def update_scheduler_data_label(scheduler_data: SchedulerData) -> None:
+    async with docker_client() as client:
+        # NOTE: builtin `DockerServices.update` function is very limited.
+        # Using the same pattern but updating labels
+
+        try:
+            # fetch information from service name
+            service_inspect = await client.services.inspect(scheduler_data.service_name)
+            service_version = service_inspect["Version"]["Index"]
+            service_id = service_inspect["ID"]
+            spec = service_inspect["Spec"]
+
+            # allows to use json encodes not available on dict
+            dict_scheduler_data = json.loads(scheduler_data.json())
```

Comment:
We have discussed about this many times now ... please check and use what is inside ``models_library.utils.fastapi_encoders``

Reply:
Thanks for pointing this out. It would be helpful in the future.
For now I've properly fixed the code here. it should be much better.